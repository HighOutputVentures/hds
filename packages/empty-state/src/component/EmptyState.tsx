import { AddIcon } from '@chakra-ui/icons';
import { Box, Flex, Icon, SystemStyleObject, Text } from '@chakra-ui/react';
import { Button } from '@highoutput/hds-forms';
import { FC, ReactElement } from 'react';
import CloudIcon from './CloudIcon';

export type EmptyStateProps = Omit<SystemStyleObject, 'onSubmit'> & {
  icon?: ReactElement;
  header: string;
  subHeader: string;
  hasPrimaryButton?: boolean;
  primaryButtonLabel?: string;
  primaryButtonIcon?: JSX.Element;
  hasSecondaryButton?: string;
  secondaryButtonLabel?: string;
  onClear?(): void;
  onSubmit?(): void;
};

const EmptyState: FC<EmptyStateProps> = (props) => {
  const {
    icon = <Icon as={CloudIcon} w="220px" h="160px" />,
    header,
    subHeader,
    hasPrimaryButton = true,
    hasSecondaryButton = true,
    primaryButtonLabel = 'Clear Search',
    primaryButtonIcon = <AddIcon w={3} h={3} />,
    secondaryButtonLabel = 'New Project',
    onClear,
    onSubmit,
    ...others
  } = props;

  return (
    <Box sx={others}>
      <Flex
        direction="column"
        justifyContent="center"
        textAlign="center"
        alignItems="center"
        padding={{
          base: 4,
          lg: 0,
        }}
      >
        <Box
          pb={{
            base: 6,
            lg: 8,
          }}
        >
          {icon}
        </Box>

        <Text
          size={{
            lg: 'label-sm-default',
            md: 'paragraph-sm-default',
          }}
          fontWeight={500}
        >
          {header}
        </Text>

        <Text
          mt={{
            base: 1,
            lg: 2,
          }}
          w={{ md: 352 }}
          size="paragraph-sm-default"
          color="Gray.500"
        >
          {subHeader}
        </Text>

        <Flex gap={4} mt={8}>
          {hasSecondaryButton && (
            <Button variant="outline" accent="gray" onClick={onClear}>
              {secondaryButtonLabel}
            </Button>
          )}

          {hasPrimaryButton && (
            <Button leftIcon={primaryButtonIcon} onClick={onSubmit}>
              {primaryButtonLabel}
            </Button>
          )}
        </Flex>
      </Flex>
    </Box>
  );
};

export default EmptyState;
