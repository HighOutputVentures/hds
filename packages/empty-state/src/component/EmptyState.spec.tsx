import { render } from '@testing-library/react';
import EmptyState from './EmptyState';

describe('EmptyState', () => {
  it('should render EmptyState', () => {
    const { container } = render(
      <EmptyState
        header={`No projects found`}
        subHeader="Your search “Landing page design” did not match any projects. Please try again."
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render EmptyState', () => {
    const { getByText } = render(
      <EmptyState
        header={`No projects found`}
        subHeader="Your search “Landing page design” did not match any projects. Please try again."
      />,
    );

    expect(getByText(`No projects found`)).toBeInTheDocument();
    expect(
      getByText(
        `Your search “Landing page design” did not match any projects. Please try again.`,
      ),
    ).toBeInTheDocument();
  });
});
