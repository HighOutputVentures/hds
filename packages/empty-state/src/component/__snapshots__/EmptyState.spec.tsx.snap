// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState should render EmptyState 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="css-xvx9jn"
    >
      <div
        class="css-110dib4"
      >
        <svg
          class="chakra-icon css-1hwoppz"
          fill="none"
          focusable="false"
          height="160"
          viewBox="0 0 220 160"
          width="220"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="110"
            cy="80"
            fill="#DDBBFE"
            r="80"
          />
          <circle
            cx="26"
            cy="20"
            fill="#C188F9"
            r="8"
          />
          <circle
            cx="198"
            cy="126"
            fill="#C188F9"
            r="6"
          />
          <circle
            cx="25"
            cy="138"
            fill="#C188F9"
            r="10"
          />
          <circle
            cx="210"
            cy="46"
            fill="#C188F9"
            r="10"
          />
          <circle
            cx="191"
            cy="11"
            fill="#C188F9"
            r="7"
          />
          <g
            filter="url(#filter0_dd_1182_2498)"
          >
            <path
              clip-rule="evenodd"
              d="M113.486 16C96.7495 16 81.9448 24.2701 72.9354 36.9466C69.9934 36.2528 66.9253 35.8857 63.7714 35.8857C41.8063 35.8857 24 53.692 24 75.6571C24 97.6223 41.8063 115.429 63.7714 115.429H163.2C182.42 115.429 198 99.8481 198 80.6286C198 61.4091 182.42 45.8286 163.2 45.8286C161.835 45.8286 160.488 45.9072 159.164 46.0601C151.546 28.3784 133.961 16 113.486 16Z"
              fill="#F3F3FF"
              fill-rule="evenodd"
            />
            <circle
              cx="63.7714"
              cy="75.6572"
              fill="url(#paint0_linear_1182_2498)"
              r="39.7714"
            />
            <circle
              cx="113.486"
              cy="65.7143"
              fill="url(#paint1_linear_1182_2498)"
              r="49.7143"
            />
            <circle
              cx="163.2"
              cy="80.6286"
              fill="url(#paint2_linear_1182_2498)"
              r="34.8"
            />
          </g>
          <g
            filter="url(#filter1_b_1182_2498)"
          >
            <rect
              fill="#7B6AD3"
              fill-opacity="0.4"
              height="56"
              rx="28"
              width="56"
              x="82"
              y="84"
            />
            <path
              d="M120.5 122.5L116.417 118.417M119.333 111.417C119.333 116.893 114.893 121.333 109.417 121.333C103.94 121.333 99.5 116.893 99.5 111.417C99.5 105.94 103.94 101.5 109.417 101.5C114.893 101.5 119.333 105.94 119.333 111.417Z"
              stroke="white"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </g>
          <defs>
            <filter
              color-interpolation-filters="sRGB"
              filterUnits="userSpaceOnUse"
              height="139.429"
              id="filter0_dd_1182_2498"
              width="214"
              x="4"
              y="16"
            >
              <feflood
                flood-opacity="0"
                result="BackgroundImageFix"
              />
              <fecolormatrix
                in="SourceAlpha"
                result="hardAlpha"
                type="matrix"
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
              />
              <femorphology
                in="SourceAlpha"
                operator="erode"
                radius="4"
                result="effect1_dropShadow_1182_2498"
              />
              <feoffset
                dy="8"
              />
              <fegaussianblur
                stdDeviation="4"
              />
              <fecolormatrix
                type="matrix"
                values="0 0 0 0 0.0627451 0 0 0 0 0.0941176 0 0 0 0 0.156863 0 0 0 0.03 0"
              />
              <feblend
                in2="BackgroundImageFix"
                mode="normal"
                result="effect1_dropShadow_1182_2498"
              />
              <fecolormatrix
                in="SourceAlpha"
                result="hardAlpha"
                type="matrix"
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
              />
              <femorphology
                in="SourceAlpha"
                operator="erode"
                radius="4"
                result="effect2_dropShadow_1182_2498"
              />
              <feoffset
                dy="20"
              />
              <fegaussianblur
                stdDeviation="12"
              />
              <fecolormatrix
                type="matrix"
                values="0 0 0 0 0.0627451 0 0 0 0 0.0941176 0 0 0 0 0.156863 0 0 0 0.08 0"
              />
              <feblend
                in2="effect1_dropShadow_1182_2498"
                mode="normal"
                result="effect2_dropShadow_1182_2498"
              />
              <feblend
                in="SourceGraphic"
                in2="effect2_dropShadow_1182_2498"
                mode="normal"
                result="shape"
              />
            </filter>
            <filter
              color-interpolation-filters="sRGB"
              filterUnits="userSpaceOnUse"
              height="72"
              id="filter1_b_1182_2498"
              width="72"
              x="74"
              y="76"
            >
              <feflood
                flood-opacity="0"
                result="BackgroundImageFix"
              />
              <fegaussianblur
                in="BackgroundImageFix"
                stdDeviation="4"
              />
              <fecomposite
                in2="SourceAlpha"
                operator="in"
                result="effect1_backgroundBlur_1182_2498"
              />
              <feblend
                in="SourceGraphic"
                in2="effect1_backgroundBlur_1182_2498"
                mode="normal"
                result="shape"
              />
            </filter>
            <lineargradient
              gradientUnits="userSpaceOnUse"
              id="paint0_linear_1182_2498"
              x1="33.2326"
              x2="103.543"
              y1="49.3796"
              y2="115.429"
            >
              <stop
                stop-color="#E3E3FC"
              />
              <stop
                offset="0.350715"
                stop-color="white"
                stop-opacity="0"
              />
            </lineargradient>
            <lineargradient
              gradientUnits="userSpaceOnUse"
              id="paint1_linear_1182_2498"
              x1="75.3122"
              x2="163.2"
              y1="32.8673"
              y2="115.428"
            >
              <stop
                stop-color="#E3E3FC"
              />
              <stop
                offset="0.350715"
                stop-color="white"
                stop-opacity="0"
              />
            </lineargradient>
            <lineargradient
              gradientUnits="userSpaceOnUse"
              id="paint2_linear_1182_2498"
              x1="136.479"
              x2="198"
              y1="57.6357"
              y2="115.429"
            >
              <stop
                stop-color="#E3E3FC"
              />
              <stop
                offset="0.350715"
                stop-color="white"
                stop-opacity="0"
              />
            </lineargradient>
          </defs>
        </svg>
      </div>
      <p
        class="chakra-text css-10rvbm3"
      >
        No projects found
      </p>
      <p
        class="chakra-text css-7djt43"
      >
        Your search “Landing page design” did not match any projects. Please try again.
      </p>
      <div
        class="css-8tq86l"
      >
        <button
          class="chakra-button css-1q9ht6o"
          data-testid="hds.button"
          type="button"
        >
          New Project
        </button>
        <button
          class="chakra-button css-1w22jc8"
          data-testid="hds.button"
          type="button"
        >
          <span
            class="chakra-button__icon css-1wh2kri"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-xwy4e8"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
                fill="currentColor"
              />
            </svg>
          </span>
          Clear Search
        </button>
      </div>
    </div>
  </div>
</div>
`;
